src/core/reactor.cc

```diff
@@ -4232,6 +4317,20 @@ class disk_config_params {
         return cfg;
     }
 
+    std::optional<file_discard_queue_config> generate_file_discard_queue_config(dev_t devid) noexcept {
+        auto mountpoint_it = _mountpoints.find(devid);
+        if (mountpoint_it == _mountpoints.end()) {
+            return std::nullopt;
+        }
+
+        auto& params = mountpoint_it->second;
+        if (params.discard_block_size && params.discard_requests_per_second) {
+            return file_discard_queue_config{*params.discard_block_size, *params.discard_requests_per_second};
```

Comment:
So the discard numbers in io-properties file are not node-wide, but per-shard?

Reply:
It works like that for the current state of the PR. It would be better to specify the value for the device instead of per-shard.

Reply:
Then consider the number of discard requests specified is less than the number of shards on the node. Is that possible? If yes, how to handle it?